chain/stmgr/stmgr.go

```diff
@@ -523,34 +528,52 @@ func (sm *StateManager) WaitForMessage(ctx context.Context, mcid cid.Cid) (*type
 			return
 		}
 
-		backTs = fts
-		backRcp = r
+		foundTs = fts
+		foundRcp = r
 		close(backSearchWait)
 	}()
 
+	forwardSearchWait := make(chan struct{})
+	go func() {
+		// use events to wait for message with confidence interval and timeout
+		err := sm.events.Called(
```

Comment:
The events system was made primary to track reverts over confidence level, and we don't really care about those here. It's also not atomic with the tipset we pass to searchBackForMsg

I'd use the notification channel directly - implementing the confidence tracking on top of it when we don't care about reverts after the confidence threshold is reached should be quite simple. This way we can also drop the rather ugly events adapter

Reply:
Not using events api also means that we can handle contexts correctly